feat: support [Column]

DeagleGross opened the pull request:
## Goal:
Support `[System.ComponentModel.DataAnnotations.Schema.ColumnAttribute]` as an additional way to override parameter name.
[Requested by some users (stackoverflow)](https://stackoverflow.com/questions/77073286/can-dapper-mappings-be-automated-with-attributes/77073456#77073456)

## Example:
To make override work, one need to specify both `[Column]` and `[UseColumnAttribute]` attributes, for example:
```csharp
class MyType
{
    [UseColumnAttribute]
    [Column("myOtherName")]
    public string Name { get; set; }
}
```

## Rules:
1. If `[Column]` and either `[UseColumnAttribute]` or `[UseColumnAttribute(true)]` is defined, than override will work
2. If `[UseColumnAttribute(false)]` is explicitly specified, then overriding will not happen
3. If `[Column]` is specified, but `[UseColumnAttribute]` is missing, then **DAP043** will be emitted to the user: "Attach the [UseColumnAttribute] attribute to make Dapper consider [Column]"
4. If _1)_ is followed, but also `[DbValue]` with property `Name` is specified, then a warning message will be emitted to the user: **DAP042** "A column name is specified via both [DbValue] and [Column]; '<dbvalue_name>' will be used"

Closes #36

mgravell commented on src/Dapper.AOT.Analyzers/Internal/Inspection.cs:
```diff
@@ -800,7 +919,7 @@ private static bool TryGetAttributeValue<T>(AttributeData? attrib, string name,
                 int index = 0;
                 foreach (var p in ctor.Parameters)
                 {
-                    if (StringComparer.InvariantCultureIgnoreCase.Equals(p.Name == name))
```
oof, good catch; that must have been fun to find!

DeagleGross replied:
you need to thank unit testing for such a treasure ;)

mgravell commented on src/Dapper.AOT/UseColumnAttributeAttribute.cs:
```diff
@@ -0,0 +1,15 @@
+    /// <summary>
+    /// Specifies whether to use [System.ComponentModel.DataAnnotations.Schema.ColumnAttribute] for additional behavioral configuration
+    /// </summary>
+    public class UseColumnAttributeAttribute : Attribute
```
what is the allowed usage here? can it be applied at the module level, or is it only respected when attached to the same type? are more than one allowed? can we add `[AttributeUsage(...)]` here accordingly

DeagleGross replied:
added such an expression.
`[AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Parameter, AllowMultiple = false)]`

let me know if such `AttributeUsage` is correct

mgravell commented:
looking good; fixed some minor nits:

- avoid `/en-us/` in docs links; if you cut it, it will use the user's locale settings
- formatting in docs
- used a different variant of that string equality fix

an open question on the new attribute, but: looking good!

DeagleGross commented:
> looking good; fixed some minor nits:
> 
> * avoid `/en-us/` in docs links; if you cut it, it will use the user's locale settings
> * formatting in docs
> * used a different variant of that string equality fix
> 
> an open question on the new attribute, but: looking good!

Thanks for the useful advices and fixed, Marc! Sorry for attribute usage - I added it. Ready for review
